src/Accordion/AccordionSection.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import AccordionHeader from './AccordionHeader';
+import AccordionPanel from './AccordionPanel';
+
+class AccordionSection extends React.Component {
+  state = {
+    isOpen: false
+  };
+
+  get isControlled() {
+    const { isOpen } = this.props;
+
+    return isOpen !== undefined;
+  }
+
+  get isOpen() {
+    const { isOpen: controlledIsOpen } = this.props;
+    const { isOpen: internalIsOpen } = this.state;
+
+    return this.isControlled ? controlledIsOpen : internalIsOpen;
```

Comment:
is this essentially `this.props.isOpen || this.state.isOpen`?

Reply:
The open state can be managed automatically or it can be managed manually. If we are manually controlling whether or not the accordion is open, we want to leverage the isOpen property. If the accordion is managed automatically we leverage it's isOpen state.

Reply:
Totally. I was just saying that this could become a one-liner `return this.props.isOpen || this.state.isOpen`. This is fine too, i guess yours is clearer.